### PR TITLE
chore(memeland): add sorting condition

### PR DESCRIPTION
**Pull request by leohhhn**
## Description

Added a few checks to the sorting of posts for Memeland. 

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: examples/gno.land/p/demo/memeland/memeland.gno [hunk 1]
```diff
@@ -122,6 +127,8 @@ func (m *Memeland) GetPostsInRange(startTimestamp, endTimestamp int64, page, pag
 			},
 		}
 		sort.Sort(dateSorter)
+	default:
+		panic("sortBy can only be \"UPVOTES\" or \"DATE_CREATED\"")
```
Comment (zivkovicmilos) [inline examples/gno.land/p/demo/memeland/memeland.gno:131]: This panic message shouldn't mention the param name. Consider rephrasing it:
`panic("sort order can only be ...")`

File: examples/gno.land/p/demo/memeland/memeland.gno [hunk 2]
```diff
@@ -113,7 +118,7 @@ func (m *Memeland) GetPostsInRange(startTimestamp, endTimestamp int64, page, pag
 			},
 		}
 		sort.Sort(dateSorter)
-	default:
+	case "DATE_CREATED":
```
Comment (zivkovicmilos) [inline examples/gno.land/p/demo/memeland/memeland.gno:121]: Why not define these as constants?

File: examples/gno.land/p/demo/memeland/memeland.gno [hunk 3]
```diff
@@ -91,6 +91,11 @@ func (m *Memeland) GetPostsInRange(startTimestamp, endTimestamp int64, page, pag
 		panic("page size cannot be larger than 10")
 	}
 
+	// Need to pass in a sort parameter
+	if sortBy == "" {
+		panic("sortBy cannot be empty")
```
Comment (zivkovicmilos) [inline examples/gno.land/p/demo/memeland/memeland.gno:96]: Please don't use the actual param name in the panic message

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

Please make a cross PR to the [memeland repo](https://github.com/gnolang/memeland)

File: examples/gno.land/p/demo/memeland/memeland.gno (hunk 3 above)
Comment (leohhhn) [reply examples/gno.land/p/demo/memeland/memeland.gno]: Fixed: [015902f](https://github.com/gnolang/gno/pull/1817/commits/015902f694b808371b3cfc3494a5423c465b5cfa)

File: examples/gno.land/p/demo/memeland/memeland.gno (hunk 2 above)
Comment (leohhhn) [reply examples/gno.land/p/demo/memeland/memeland.gno]: Defined: [015902f](https://github.com/gnolang/gno/pull/1817/commits/015902f694b808371b3cfc3494a5423c465b5cfa)

File: examples/gno.land/p/demo/memeland/memeland.gno (hunk 1 above)
Comment (leohhhn) [reply examples/gno.land/p/demo/memeland/memeland.gno]: Fixed: [015902f](https://github.com/gnolang/gno/pull/1817/commits/015902f694b808371b3cfc3494a5423c465b5cfa)

**Comment by leohhhn** [comment]
@zivkovicmilos added cross PR to memeland repo: https://github.com/gnolang/memeland/pull/17

**Comment by zivkovicmilos** [comment]
@leohhhn 
Please fix the linter and we can go ahead and merge 🙏

**Pull request merged**